src/components/MagicCodeInput.js

```diff
@@ -244,24 +301,37 @@ function MagicCodeInput(props) {
             }
 
             const newFocusedIndex = Math.max(0, focusedIndex - 1);
+
+            // Saves the input string so that it can compare to the change text
+            // event that will be triggered, this is a workaround for mobile that
+            // triggers the change text on the event after the key press.
+            setInput(TEXT_INPUT_EMPTY_STATE);
+            setFocusedIndex(newFocusedIndex);
+            setEditIndex(newFocusedIndex);
```

Comment:
We use these three lines in a number of places - might it be good to separate them out into a helper function?